apps/explorer/src/components/ownedobjects/views/UnderlyingObjectCard.tsx

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useGetDynamicFieldObject } from '@mysten/core';
+import { getObjectDisplay, getObjectFields, getObjectId } from '@mysten/sui.js';
+
+import { SyntaxHighlighter } from '~/components/SyntaxHighlighter';
+import DisplayBox from '~/components/displaybox/DisplayBox';
+import { Banner } from '~/ui/Banner';
+import { ObjectLink } from '~/ui/InternalLink';
+import { LoadingSpinner } from '~/ui/LoadingSpinner';
+import { Text } from '~/ui/Text';
+import {
+    parseImageURL,
+    parseObjectType,
+    extractName,
+} from '~/utils/objectUtils';
+import { trimStdLibPrefix } from '~/utils/stringUtils';
+
+interface UnderlyingObjectCardProps {
+    parentId: string;
+    name: {
+        type: string;
+        value?: string;
+    };
+}
+
+export function UnderlyingObjectCard({
+    parentId,
+    name,
+}: UnderlyingObjectCardProps) {
+    const { data, isLoading, isError, isFetched } = useGetDynamicFieldObject(
+        parentId,
+        name
+    );
+
+    if (isLoading) {
+        return (
+            <div className="mt-3 flex w-full justify-center pt-3">
+                <LoadingSpinner text="Loading data" />
+            </div>
+        );
+    }
+
+    if (isError || data.error || (isFetched && !data)) {
+        return (
+            <Banner variant="error" spacing="lg" fullWidth>
+                Failed to get field data for :{parentId}
+            </Banner>
+        );
+    }
+    const display = getObjectDisplay(data);
+    const imgUrl = parseImageURL(display.data);
+    const objectType = parseObjectType(data);
+    const caption = extractName(display.data) || trimStdLibPrefix(objectType);
+    // For
+    const underlyingObjectTypes =
+        typeof name.value === 'object'
+            ? objectType
+                  ?.slice(objectType?.indexOf('<') + 1, objectType.indexOf('>'))
+                  .split(',')
+            : [];
+    const fieldsData = getObjectFields(data!);
+
+    return (
+        <>
+            <SyntaxHighlighter
+                code={JSON.stringify(fieldsData, null, 2)}
+                language="json"
+            />
+
+            {underlyingObjectTypes.length > 0 ? (
+                <div className="mt-3 pt-3">
+                    <Text variant="body/medium" color="steel-dark">
+                        Underlying Object
+                    </Text>
+                    <div className="item-center mt-5 flex justify-start gap-1">
+                        {imgUrl ? (
```

Comment:
Base on the current response of the api this will never show up.  dynamic_field::Field does not have a display standard implementation. And doesn't resolve or expose the display of it's value

Reply:
Yes, But I think we should keep this so whenever the displayField is added it works